lib/translator/status.cc

```diff
@@ -0,0 +1,252 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "status.h"
+
+namespace translator {
+
+namespace {  // anonymous namespace for helper functions
+
+// Section 7.1
+// https://www.nvmexpress.org/wp-content/uploads/NVM-Express-SCSI-Translation-Reference-1_1-Gold.pdf
+StatusCode Status(nvme_defs::GenericCommandStatusCode sc, ScsiStatus& result) {
```

Comment:
Should the name of this family of functions indicate that it's translating from NVMe to SCSI?

Reply:
That's a good point. As it is, it's unclear that the translation is from NVMe to SCSI, so i'll rethink the naming and consult the others about this

Reply:
Renamed to `StatusToScsi()`